### ISSUE TITLE
Exception thrown when logging in

### ISSUE THREAD

**Issue by waynegraham** [uvalib/geoblacklight#11]
When logging in with pub_cookie enabled, the session controller throws an exception when devise attempts to create a new session with the user id.  

```
F, [2015-06-08T09:33:22.744672 #13991] FATAL -- :
ArgumentError (When assigning attributes, you must pass a hash as an argument.):
  app/controllers/devise/sessions_controller.rb:10:in `new'
```

It looks like the `REMOTE_USER` isn't getting passed back:

https://github.com/waynegraham/geoblacklight/blob/develop/app/controllers/devise/sessions_controller.rb#L10

**Comment by waynegraham** [comment]
I built a branch and made some progress on this...turns out this is a nasty/fun programming issue because of how pub_cookie works. 

I added  https://github.com/waynegraham/geoblacklight/blob/devise/config/initializers/pubcookie_authentication.rb

Which mostly takes care of the hand-off. What I discovered is that the header `'REMOTE_USER'` is only passed back on the initial visit to the pubcookie auth site. The only reliable method I could think of to get the username to fake an email address is to decrypt the `pubcookie_s_geoblacklight` cookie that gets set. Unfortunately, at this point, this requires reading the key which throws an permission error on the server. 

Just so my notes are in one place, the pubcookie C code for reversing the decryption is at http://svn.cac.washington.edu/viewvc/pubcookie/trunk/src/pubcookie.h?view=markup

Of particular interest is the `PBC_INIT_IVEC` (63) for blowfish along with all the struct sizes starting at 40. If we can get read access to this file, it should be relatively straight forward to finish the `extract_username` method (and the `des_decrypt`).
